cli/src/main/java/com/devonfw/tools/ide/tool/docker/Docker.java

```diff
@@ -0,0 +1,79 @@
+package com.devonfw.tools.ide.tool.docker;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import com.devonfw.tools.ide.common.Tag;
+import com.devonfw.tools.ide.context.IdeContext;
+import com.devonfw.tools.ide.os.SystemArchitecture;
+import com.devonfw.tools.ide.repo.ToolRepository;
+import com.devonfw.tools.ide.tool.GlobalToolCommandlet;
+import com.devonfw.tools.ide.tool.PackageManager;
+import com.devonfw.tools.ide.version.VersionIdentifier;
+
+public class Docker extends GlobalToolCommandlet {
+  /**
+   * The constructor.
+   *
+   * @param context the {@link IdeContext}.
+   */
+  public Docker(IdeContext context) {
+
+    super(context, "docker", Set.of(Tag.DOCKER));
+  }
+
+  @Override
+  public boolean isExtract() {
+
+    return switch (context.getSystemInfo().getOs()) {
+      case WINDOWS -> false;
+      case MAC -> context.getSystemInfo().getArchitecture().equals(SystemArchitecture.ARM64);
+      case LINUX -> true;
+    };
+  }
+
+  @Override
+  protected boolean doInstall(boolean silent) {
+
+    if (context.getSystemInfo().isLinux()) {
+      return installWithPackageManger(getPackageMangerCommands(), silent);
+    } else {
+      return super.doInstall(silent);
+    }
+  }
+
+  private Map<PackageManager, List<String>> getPackageMangerCommands() {
+
+    Map<PackageManager, List<String>> commands = new HashMap<>();
+
+    String edition = getEdition();
+    ToolRepository toolRepository = this.context.getDefaultToolRepository();
+    VersionIdentifier configuredVersion = getConfiguredVersion();
+    String resolvedVersion = toolRepository.resolveVersion(this.tool, edition, configuredVersion).toString();
+
+    commands.put(PackageManager.ZYPPER, Arrays.asList(
+        "sudo zypper addrepo https://download.opensuse.org/repositories/isv:/Rancher:/stable/rpm/isv:Rancher:stable.repo",
+        String.format("sudo zypper --no-gpg-checks install rancher-desktop=%s*", resolvedVersion)));
+
+    commands.put(PackageManager.APT, Arrays.asList(
+        "curl -s https://download.opensuse.org/repositories/isv:/Rancher:/stable/deb/Release.key | gpg --dearmor | sudo dd status=none of=/usr/share/keyrings/isv-rancher-stable-archive-keyring.gpg",
+        "echo 'deb [signed-by=/usr/share/keyrings/isv-rancher-stable-archive-keyring.gpg] https://download.opensuse.org/repositories/isv:/Rancher:/stable/deb/ ./' | sudo dd status=none of=/etc/apt/sources.list.d/isv-rancher-stable.list",
+        "sudo apt update",
+        String.format("sudo apt install -y --allow-downgrades rancher-desktop=%s*", resolvedVersion)));
```

Comment:
OK, now I at least get why you are using `Map<PackageManager, List<String>>`.
However, can't we still keep that simple as a String vararg?
I see that when there is `curl - s ... && echo ... && sudo apt update` it seems tricky to get the package manager.
So far we had simpler situations where we just called the package-manager and that was the first argument.

My conclusion would be:
1. we are running an installation so all logic can run with a single `sudo` command so that could be omitted here and done in `installWithPackageManger` method.
2. IMHO we can combine all the individual string arguments to a single script.
3. The remaining challenge is to also provide the package manager or OS so that `installWithPackageManger` can choose which of the installation options to take.

How about a simple API for the easy 90% case:
`installWithPackageManager("apt-get install unzip", "yum install unzip", "pacman -S unzip", "emerge install unzip", "zypp install unzip", "apk add unzip", "brew install unzip");`
Here we would extract the package manager directly from the command and convert it to our `Map` structure.
We can even make it smart to search in the string for any of the known package-managers if not already the first word is a package manager.
However, to avoid too much magic in really complex cases, we can still call a more complex method giving structured data instead of just Strings.
But still I think that `Map` is not the best option. What we want to provide is a "list" of "installation variants" per OS/Distribution/Package-Manager. For this a vararg of `PackageManagerCommand` would make more sense.
The class `PackageManagerCommand` can have the `PackageManager` and again a list of commands to keep the existing flexibility. A simple `PackageManagerCommand.of("apt install unzip")` would set `PackageManager` to `apt` and the commands list to the single string `apt install unzip`.

Reply:
Maybe from security PoV and zero trust we should question my point 1.